From 1be6161c596c6bb6eb537e6027b1b4972f397e88 Mon Sep 17 00:00:00 2001
From: Umang Chheda <quic_uchheda@quicinc.com>
Date: Wed, 18 Oct 2023 18:14:15 +0530
Subject: [PATCH 20/27] PENDING: arm64: dts: qcm6490-rb3: Remove voltage voting
 for USB rails

USB driver does not vote for voltage on hsphy and ssphy
rails. Due to which the initial voltage set by bootloader
is overridden by regulator framework with min voltage specified
on regulator registration.

Fix this temporarily by removing voltage voting support, which
will prevent regulator framework overriding the voltage set by
bootloader for QC6490 RB3 Platform.

This commit will be reverted once voltage voting support is added
in USB driver.

Change-Id: I5bf30f9ac8b58ac60cdabbc756775bf3be679e9b
Signed-off-by: Umang Chheda <quic_uchheda@quicinc.com>
Signed-off-by: Atul Dhudase <quic_adhudase@quicinc.com>
Upstream-Status: Pending
---
 arch/arm64/boot/dts/qcom/qcm6490-rb3.dts | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/qcm6490-rb3.dts b/arch/arm64/boot/dts/qcom/qcm6490-rb3.dts
index 3a1c781c965f..b244e66e9857 100644
--- a/arch/arm64/boot/dts/qcom/qcm6490-rb3.dts
+++ b/arch/arm64/boot/dts/qcom/qcm6490-rb3.dts
@@ -338,6 +338,13 @@ &usb_1_qmpphy {
 	status = "okay";
 };
 
+&vreg_l1b_0p912 {
+	/delete-property/regulator-min-microvolt;
+	/delete-property/regulator-max-microvolt;
+	regulator-allow-set-load;
+	regulator-allowed-modes = <RPMH_REGULATOR_MODE_LPM RPMH_REGULATOR_MODE_HPM>;
+};
+
 &vreg_l7b_2p952 {
 	/delete-property/regulator-min-microvolt;
 	/delete-property/regulator-max-microvolt;
@@ -351,3 +358,10 @@ &vreg_l9b_1p2 {
 	regulator-allow-set-load;
 	regulator-allowed-modes = <RPMH_REGULATOR_MODE_LPM RPMH_REGULATOR_MODE_HPM>;
 };
+
+&vreg_l10c_0p88 {
+	/delete-property/regulator-min-microvolt;
+	/delete-property/regulator-max-microvolt;
+	regulator-allow-set-load;
+	regulator-allowed-modes = <RPMH_REGULATOR_MODE_LPM RPMH_REGULATOR_MODE_HPM>;
+};
-- 
2.25.1

